src/output.js

```diff
@@ -0,0 +1,392 @@
+ */
+async function run({ status }) {
+  try {
+    const {
+      slackWebhookUrl,
+      applicationGithubToken,
+      personalGithubToken,
+      githubNotification: githubNotificationEnabled,
+      slackNotification: slackNotificationEnabled
+    } = input
+    const shouldRunOutput = input.logLevel === 'info' || (input.logLevel === 'error' && status)
+
+    if (!shouldRunOutput) {
+      return Promise.resolve()
+    }
+
+    const slackEnabled = slackNotificationEnabled && slackWebhookUrl
+    const githubEnabled = githubNotificationEnabled && applicationGithubToken
+
+    /**
+     * @type {[ LHResultsByURL, ChangesURL, Gist[] ]}
+     */
+    const [groupedResults, changesURL, gists] = await Promise.all([
+      getGroupedAssertionResultsByURL(),
+      getChangesUrl({ githubToken: personalGithubToken }),
+      // keep uploading as part of Promise all instead of separate request
+      uploadResultsToGist({ githubToken: personalGithubToken })
+    ])
+
+    const slackData = { status, slackWebhookUrl, changesURL, gists, groupedResults }
+    const githubData = { status, githubToken: applicationGithubToken, changesURL, gists, groupedResults }
+
+    if (githubEnabled && slackEnabled) {
+      await Promise.all([slackNotification(slackData), githubNotification(githubData)])
+    } else if (githubEnabled) {
+      await githubNotification(githubData)
+    } else if (slackEnabled) {
+      await slackNotification(slackData)
+    } else {
+      // @todo log notification and link to doc?
+    }
+  } catch (e) {
+    console.log(e)
+    throw e
+  }
+}
+
+/**
+ * @param {{status: number, slackWebhookUrl?: string, changesURL: ChangesURL, gists: Gist[], groupedResults: LHResultsByURL }} params
+ * @return {Promise<*>}
+ */
+async function slackNotification({ status, slackWebhookUrl = '', changesURL, groupedResults, gists }) {
+  console.log('Running Slack notification')
+
+  const webhook = new IncomingWebhook(slackWebhookUrl)
+  const color = status === 0 ? 'good' : 'danger'
+  const conclusion = status === 0 ? 'success' : 'failure'
+  const changesTitle = changesURL.pullRequest
+    ? `Pull Request ${conclusion} - <${changesURL.pullRequest} | View on GitHub>`
+    : `Changes ${conclusion} - <${changesURL.sha} | View SHA Changes>`
+  const attachments = formatAssertResults({ groupedResults, status, gists })
+  return webhook.send({
+    attachments: [
+      {
+        pretext: `GitHub Actions / ${reportTitle}`,
+        title: changesTitle,
+        color
+      },
+      ...attachments
+    ]
+  })
+}
+
+/**
+ * @param {{status: number, githubToken?: string, changesURL: ChangesURL, gists: Gist[], groupedResults: LHResultsByURL }} params
+ * @return {Promise<*>}
+ */
+async function githubNotification({ status, githubToken = '', changesURL, gists, groupedResults }) {
+  console.log('Running Github notification')
+
+  const conclusion = status === 0 ? 'success' : 'failure'
+  const octokit = new github.GitHub(githubToken)
```

Comment:
The run logs were saying something about the `octokit` client being weird. I couldn't really tell that this (github noti) was working, I didn't know what check to look at to confirm?

Reply:
I tested it mostly using real github action for test repo - https://github.com/test-denar/test-action/actions/runs/36681044
`applicationGithubToken` requires application token, so I managed to do that only with real repo, for now...
this is one more thing to improve for developers/contributors experience.